metaflow/plugins/kfp/tests/run_integration_tests.py

```diff
@@ -100,19 +103,60 @@ def test_s3_sensor_flow(pytestconfig) -> None:
     return
 
 
-# this test ensures the integration tests fail correctly
-def test_raise_failure_flow(pytestconfig) -> None:
+# This test ensures the integration tests fail correctly
+# and that when a test fails, an OpsGenie email is sent.
+# It verifies the OpsGenie email has been sent by querying
+# the OpsGenie API to look for the corresponding alert
+# and then closes that alert.
+def test_error_and_opsgenie_alert(pytestconfig) -> None:
     test_cmd = (
         f"{_python()} flows/raise_error_flow.py --datastore=s3 kfp run "
         f"--wait-for-completion --workflow-timeout 1800 "
-        f"--experiment metaflow_test --tag test_t1 "
+        f"--experiment metaflow_test --tag test_t1 --notify "
     )
     if pytestconfig.getoption("image"):
         test_cmd += (
             f"--no-s3-code-package --base-image {pytestconfig.getoption('image')}"
         )
 
-    exponential_backoff_from_platform_errors(test_cmd, 1)
+    kfp_run_id = exponential_backoff_from_platform_errors(test_cmd, 1)
+    # TODO use @secret (when it's released) to store this API key
+    opsgenie_auth_headers = {
+        "Content-Type": "application/json",
+        "Authorization": "GenieKey 21bda224-d719-42e5-aceb-35616eb26a6c",
```

Comment:
Can you please remove this from the repo. Having an API key in a public repo really isn't acceptable and we'll need to think through how best to store this unfortunately.

Reply:
Can we store this in a Gitlab secret and when these tests are run via Gitlab we can pull this secret in via the CLI command in the `.gitlab-ci.yml`?

Reply:
Got it, my bad! I just regenerated the key, so this key should no longer pose a threat.

Reply:
@alexlatchford thanks for this quick catch! I updated this PR to use a Gitlab "masked" variable so it's not revealed in plaintext. I also updated to using a new API key.